### PR TITLE
Fix player does't use the Shellphone (Ocean), Shellphone (Underworld)…

### PR DESCRIPTION
**Pull request by hufang360**
Fix player does't use the Shellphone (Ocean), Shellphone (Underworld) and Shellphone (Spawn)

### CODE REVIEW

File: TShockAPI/GetDataHandlers.cs [hunk 1]
```diff
@@ -3828,31 +3828,56 @@ void Fail(string tpItem)
...
+						args.Player.SelectedItem.type != ItemID.ShellphoneOcean)
 					{
 						TShock.Log.ConsoleDebug("GetDataHandlers / HandleTeleportationPotion rejected not holding the correct item {0} {1}", args.Player.Name, type);
 						return true;
 					}
 
 					if (!args.Player.HasPermission(Permissions.magicconch))
 					{
-						Fail("the Magic Conch");
+						if (args.Player.ItemInHand.type == ItemID.ShellphoneOcean || args.Player.SelectedItem.type != ItemID.ShellphoneOcean)
```
Comment (drunderscore) [inline TShockAPI/GetDataHandlers.cs:3843]: This check doesn't seem right, instead it should be:
```cs
if (args.Player.ItemInHand.type == ItemID.ShellphoneOcean || args.Player.SelectedItem.type == ItemID.ShellphoneOcean)
```

File: TShockAPI/GetDataHandlers.cs [hunk 2]
```diff
@@ -3828,31 +3828,56 @@ void Fail(string tpItem)
...
+						args.Player.SelectedItem.type != ItemID.ShellphoneHell)
 					{
 						TShock.Log.ConsoleDebug("GetDataHandlers / HandleTeleportationPotion rejected not holding the correct item {0} {1}", args.Player.Name, type);
 						return true;
 					}
 
 					if (!args.Player.HasPermission(Permissions.demonconch))
 					{
-						Fail("the Demon Conch");
+						if (args.Player.ItemInHand.type == ItemID.ShellphoneHell || args.Player.SelectedItem.type != ItemID.ShellphoneHell)
```
Comment (drunderscore) [inline TShockAPI/GetDataHandlers.cs:3866]: Similar to above, this should be:
```csharp
if (args.Player.ItemInHand.type == ItemID.ShellphoneHell || args.Player.SelectedItem.type == ItemID.ShellphoneHell)
```

**Review by drunderscore — COMMENTED** [review]
As long as the suggested changes are made, this works totally fine :+1:

**Comment by hufang360** [comment]
> As long as the suggested changes are made, this works totally fine 👍

Thank you! I forget modify the CHANGELOG.md. Now, I pull it agrain.

**Pull request closed without merging**